File: Persistencia/Scripts/micro_dao/servico_subtipo_servico.py
from micro_dao.objeto_modelo import ObjetoModelo


class ServicoSubtipoServico(ObjetoModelo):

    _id: int
    _fk_servico: int
    _fk_subtipo_servico: int

    def __init__(self, fk_servico: int, fk_subtipo_servico: int):
        self.set_fk_servico(fk_servico)
        self.set_fk_subtipo_servico(fk_subtipo_servico)


    def get_id(self) -> int:
        return self._id

    def set_id(self, id: int):
        self._id = id
        return self

    def set_fk_servico(self, fk_servico: int):
        self._fk_servico = fk_servico
        return self

    def get_fk_servico(self) -> int:
        return self._fk_servico

    def set_fk_subtipo_servico(self, fk_subtipo_servico: int):
        self._fk_subtipo_servico = fk_subtipo_servico
        return self

    def get_fk_subtipo_servico(self) -> int:
        return self._fk_subtipo_servico

File: Projeto_Java/src/Dominio/Entidades/ServicoSubtipoServico.java
package Dominio.Entidades;

import javax.persistence.*;

@Entity
@Table(name = "servico_subtipo_servico")
public class ServicoSubtipoServico implements Comparable<ServicoSubtipoServico> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "fk_servico")
    private int fkServico;

    @Column(name = "fk_subtipo_servico")
    private int fkSubtipoServico;


    public ServicoSubtipoServico() {};

    public ServicoSubtipoServico(int fkServico, int fkSubtipoServico) {
        this.setFkServico(fkServico);
        this.setFkSubtipoServico(fkSubtipoServico);
    }

    public ServicoSubtipoServico(int id, int fkServico, int fkSubtipoServico) {
        this.setId(id);
        this.setFkServico(fkServico);
        this.setFkSubtipoServico(fkSubtipoServico);
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getFkServico() {
        return fkServico;
    }

    public void setFkServico(int fkServico) {
        this.fkServico = fkServico;
    }

    public int getFkSubtipoServico() {
        return fkSubtipoServico;
    }

    public void setFkSubtipoServico(int fkSubtipoServico) {
        this.fkSubtipoServico = fkSubtipoServico;
    }

    @Override
    public int compareTo(ServicoSubtipoServico servicoSubtipoServico) {
        return Integer.compare(this.getId(), servicoSubtipoServico.getId());
    }
}
